### PR TITLE
Hydromancer new E, Dreadlord spell fixes, New Ghoul spells

### PR DESCRIPTION
**Pull request by SebastianTanoni**

### CODE REVIEW

**Comment by YakaryBovine** [comment]
image
Do you want the Ghoul to have this little bright orb on his hand?

**Comment by SebastianTanoni** [comment]
> image Do you want the Ghoul to have this little bright orb on his hand?

thats a good question. Do you think the passive is fine without a visual aid on the unit?

**Comment by YakaryBovine** [comment]
> > image Do you want the Ghoul to have this little bright orb on his hand?
> 
> thats a good question. Do you think the passive is fine without a visual aid on the unit? Since you worry more about visual clarity, i think you are better to answer! i can remove it if you think its unneded

Yeah I don't think it's needed; the light bulb doesn't really mean anything to players anyway, especially since no other unit shares it. I'd also be worried about special effect lag considering Ghouls can be spammed.

BTW I had to run to work before I could finish the review, that's not the whole thing 😆

**Comment by SebastianTanoni** [comment]
Perfect, il remove the effect!

File: mapdata/WarcraftLegacies/AbilityData/Skin/1431515201.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    {
+      "Id": 1953653089,
+      "Type": 3,
+      "Value": "ReplaceableTextures\\PassiveButtons\\PASDWRip.blp"
+    },
+    {
+      "Level": 1,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Increases damage dealt to enemy targets with unarmored armor by \u003CA0SU,DataB1,%\u003E%."
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1431515201.json:47]:
```suggestion
      "Value": "Increases damage dealt to enemy targets with Unarmored armor by \u003CA0SU,DataB1,%\u003E%."
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1465069633.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+      "Id": 1718840929,
+      "Type": 3,
+      "Value": "(scourge ghoul)"
+    },
+    {
+      "Level": 1,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Causes the Ghoul to keep fighting upon death, allowing it to continue fighting for 30 seconds. The Ghoul will be uncontrollable and invulnerable."
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1465069633.json:17]:
```suggestion
      "Value": "Upon death, the Ghoul becomes invulnerable and uncontrollable and continues fighting for 30 seconds."
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/959590465.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
       "Level": 1,
       "Id": 828536161,
       "Type": 3,
-      "Value": "Summons a water elemental. Has \u003Ch00U,realHP\u003E hit points and deals \u003Ch00U,mindmg1\u003E - \u003Ch00U,maxdmg1\u003E damage. |nLasts \u003CA029,Dur1\u003E seconds. Elemental are affected by Power Infusion upgrades. |n|n|cffffcc00Attacks land and air units.|r"
+      "Value": "Summons a water elemental. Has \u003Ch00U,realHP\u003E hit points and deals \u003Ch00U,mindmg1\u003E - \u003Ch00U,maxdmg1\u003E damage. Can Learn the Freezing Shot ability. |nLasts \u003CA029,Dur1\u003E seconds. Elemental are affected by Power Infusion upgrades and the Hydromancer caster training.|n|n|cffffcc00Attacks land and air units.|r"
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/959590465.json:23]:
```suggestion
      "Value": "Summons a Water Elemental, which has \u003Ch00U,realHP\u003E hit points, deals \u003Ch00U,mindmg1\u003E - \u003Ch00U,maxdmg1\u003E damage, and can learn the Freezing Shot ability. |nLasts \u003CA029,Dur1\u003E seconds. Water Elementals are affected by the Power Infusion and Hydromancer Training researches.|n|n|cffffcc00Attacks land and air units.|r"
```

File: mapdata/WarcraftLegacies/AbilityData/Skin/1278488641.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      "Level": 3,
+      "Id": 829453409,
+      "Type": 3,
+      "Value": "Black Hole - [|cffffcc00Level 3|r]"
+    },
+    {
+      "Level": 1,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Summon a Storm cloud at a target location. The Storm cloud will fire a lightning bolt every 2 seconds, dealing 50 damage to a random enemy. In addition, the Storm Cloud will deal \u003CA0CZ,DataA1\u003E damage per second to nearby enemy air units.|nLasts \u003CA04L,Dur1\u003E seconds, has a \u003CA04L,Cast1\u003E seconds cast time and is channeled."
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1278488641.json:44]:
```suggestion
      "Value": "Channels a Storm Cloud at the target location after casting for \u003CA04L,Cast1\u003E seconds. The Storm Cloud will fire a lightning bolt every 2 seconds, dealing 50 damage to a random enemy. In addition, the Storm Cloud will deal \u003CA0CZ,DataA1\u003E damage per second to nearby enemy air units.|nLasts \u003CA04L,Dur1\u003E seconds."
```

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Just some minor tooltip stuff to fix. Great otherwise.

Personally I'm not a fan of Undying Frenzy. I don't think a tier 1 unit should have 5 abilities, especially when none of them are gated behind researches. Between Ripping Claw, Charge, and Skeletal Frame, surely the unit has a concrete identity. 
The implementation is a bit iffy as well. Though the half-Ghoul model looks great and the uncontrollable aspect is thematic, because the unit still _looks_ like a unit, the mechanic just comes across as buggy-feeling to me. Does it need to be untargetable to work?
Your call on the ability anyway, just offering 2 cents. Only need to fix the tooltip issues for me to merge.

**Comment by SebastianTanoni** [comment]
Todo: remove skeletal frame, remove locust aspect from undying frenzy

**Comment by YakaryBovine** [comment]
Also, only just thought to check: Ripping Claws doesn't work. Tried attacking some Unarmored casters and it doesn't seem to change the Ghoul's damage.

**Comment by YakaryBovine** [comment]
I double-checked the corpse thing for you; the answer is that neither the Ghoul nor the reanimated Ghoul leaves a corpse. The `SummonUnitOnDeath` trigger currently removes the dying unit, and I guess summoned units never leave behind corpses. So no corpse.

File: mapdata/WarcraftLegacies/AbilityData/Skin/1278488641.json [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      "Level": 3,
+      "Id": 829453409,
+      "Type": 3,
+      "Value": "Black Hole - [|cffffcc00Level 3|r]"
+    },
+    {
+      "Level": 1,
+      "Id": 828536161,
+      "Type": 3,
+      "Value": "Channels a Storm Cloud at the target location after casting for \u003CA04L,Cast1\u003E seconds. The Storm Cloud will attack a random enemy in range, dealing \u003Cn03J,maxdmg1\u003E chaos damage. In addition, the Storm Cloud will deal \u003CA0CZ,DataA1\u003E damage per second to nearby enemy air units.|nLasts \u003CA04L,Dur1\u003E seconds."
```
Comment (YakaryBovine) [inline mapdata/WarcraftLegacies/AbilityData/Skin/1278488641.json]:
```suggestion
      "Value": "Channels a Storm Cloud at the target location after casting for \u003CA04L,Cast1\u003E seconds. The Storm Cloud will attack nearby enemy units, dealing \u003Cn03J,maxdmg1\u003E Chaos damage with each attack. In addition, the Storm Cloud will deal \u003CA0CZ,DataA1\u003E damage per second to nearby enemy air units.|nLasts \u003CA04L,Dur1\u003E seconds."
```

**Review by YakaryBovine — APPROVED** [review]
Looking good!

**Pull request merged**